backend/test/resolver.spec.js

```diff
@@ -0,0 +1,91 @@
+const { gql } = require('apollo-server-express');
+const { createTestClient } = require('apollo-server-testing');
+const { getTestApolloServer } = require('../server.js');
+const jwt = require('jsonwebtoken');
+const server = getTestApolloServer();
+const initClient = createTestClient(server);
+const query = initClient.query;
+const mutate = initClient.mutate;
+
+// could not outsource these queries/mutations...
+const ALL_ITEMS_QUERY = gql`
+    query itemsQuery {
+        items {
+            id,
+            text,
+            editMode
+        }
+    }
+`;
+
+const CREATE_ENTRY = gql`
+    mutation createEntry($text: String!) {
+        createEntry(text: $text) {
+            id,
+            text,
+            editMode
+        }
+    }
+`;
+
+const DELETE_ENTRY = gql`
+    mutation DeleteEntry($id: String!) {
+        deleteEntry(id: $id)
+    }
+`;
+
+const LOGIN = gql`
+    mutation login($email: String!, $password:String!) {
+        login(email: $email, password: $password)
+    }
+`;
+
+describe('User is logged in', () => {
+    describe('Queries', () => {
+        it("has start todo items", () => {
+            query({ query: ALL_ITEMS_QUERY }).then((data) => {
+                expect(data.data.items).toHaveLength(4);
+                expect(data.data.items).toMatchObject([
```

Comment:
I think the two expectations are redundant.

I would the test case with `async/await` syntax:
```js
describe('todos', () => {
  it("responds with all todos", async () => {
    await expect(query({ query: ALL_ITEMS_QUERY })).resolves.toMatchObject({
      data: {
        todos: [
          { id: 1 },
          { id: 2 },
          { id: 3 },
          { id: 4 },
        ]
      },
      errors: undefined,
    })
  })
})
```